File: dtc-flink-zabbix/src/main/scala/com/dtc/java/analytic/V1/common/utils/DtcConf.java
package com.dtc.java.analytic.V1.common.utils;

import com.dtc.java.analytic.V1.dtc.execption.DtcException;
import org.apache.hadoop.conf.Configuration;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.*;

/**
 * Created on 2019-08-16
 *
 * @author :hao.li
 */
public class DtcConf {
    private static Logger logger = LoggerFactory.getLogger(DtcConf.class);
    private static Configuration conf = new Configuration();

    /**
     * Initial or refresh configuration & constants. Before you starting
     * applications on Hadoop/HBase, you should call this method at first. If you
     * are testing units, calling this method might not be necessary.
     */
    public static void setup() {
        InputStream inputStream = null;
        try {
            inputStream = DtcConf.class.getClassLoader().getResourceAsStream("dtc-flink.xml");
//            inputStream = ClassLoader.getSystemResourceAsStream("dtc-flink.xml");
            //local test
//            inputStream = new FileInputStream(new File("conf/dtc-flink.xml"));
            if (null != inputStream) {
//        conf.set("hadoop.tmp.dir", "/tmp");
//        conf.set("fs.defaultFS", "hdfs://10.3.6.7:9000");
                conf.addResource(inputStream);
                String kafka=conf.get("dtc.bootstrap.servers");
            } else {
                throw new IOException("Configure file not found: dtc-flink.xml");
            }
        } catch (FileNotFoundException e) {
            logger.error("An error was caught", e);
        } catch (IOException e) {
            throw new DtcException(e.getMessage());
        } finally {
            try {
                if (inputStream != null) {
                    inputStream.close();
                }
            } catch (Exception e) {
                logger.error("An error was caught", e);
            }
        }
    }


    public static Configuration getConf() {
        return conf;
    }
}

File: dtc-flink-zabbix/src/main/scala/com/dtc/scala/analytic/common/DtcConf.scala
package com.dtc.scala.analytic.common

import java.io.{FileNotFoundException, IOException, InputStream}

import com.dtc.scala.analytic.dtcexpection.DtcException
import org.apache.hadoop.conf.Configuration
import org.slf4j.{Logger, LoggerFactory}

/**
  * Created on 2019-05-27
  *
  * @author :hao.li
  */
object DtcConf {

  def logger: Logger = LoggerFactory.getLogger(DtcConf.getClass)

  private val conf = new Configuration()

  def setup(): Unit = {
    var inputStream: InputStream = null
    try {
//            var inputStream = new FileInputStream(new File("conf/dtc-flink.xml"))
      var inputStream = DtcConf.getClass.getClassLoader.getResourceAsStream("dtc-flink.xml")
      if (inputStream != null) {
        conf.addResource(inputStream)
      } else {
        throw new IOException("Configure file not found: dtc-flink.xml")
      }
    } catch {
      case ex: FileNotFoundException => {
        logger.error("Configure file not found: dtc-flink.xml,and the cause is {}.", ex)
        throw new DtcException("dtc-flink.xml is not found!")
      }
      case ex: IOException => {
        logger.error("Configure file not found: dtc-flink.xml,and the cause is {}.", ex)
      }

    } finally {
      if (inputStream != null) {
        inputStream.close()
      }
    }

  }

  def getConf(): Configuration = {
    return conf
  }

}
